### PR TITLE
Add package baseline

### PR DESCRIPTION
**Pull request by ericstj**
Package baseline is now defined in individual repos instead of
buildtools.

Pick up CoreFx's baseline from the Microsoft.Private.PackageBaseline
package and add a WCF specific baseline.

I've primed MyGet with the required version of this package.

Needs: https://github.com/dotnet/buildtools/pull/646

### CODE REVIEW

**Comment by ericstj** [comment]
Removed WIP tag, this is ready to be merged.  @roncain can you have a look?

File: Packaging.props [hunk 1]
```diff
@@ -16,8 +16,18 @@
     <PackagePlatform Condition="'$(PackagePlatform)' == ''">$(Platform)</PackagePlatform>
```
Comment (weshaggard) [inline Packaging.props]: orthogonal but we should also use CoreFxExpectedPrerelease for the Microsoft.NETCore.Platforms version above in this file.

**Comment by roncain** [comment]
@dotnet-bot test outerloop Windows_NT Debug, test outerloop Windows_NT Release

**Comment by roncain** [comment]
OuterLoop failures in CI appear to be due to problems in our test infrastructure.  I verified manually I can run OuterLoops on Windows.

/cc @StephenBonikowsky for his input on the changes.

**Comment by StephenBonikowsky** [comment]
This change means that Packaging.props diverges significantly from the version in corefx, is it not possible to have a unified version that does the right thing based on the repo?

**Comment by weshaggard** [comment]
@StephenBonikowsky I just chatted with @ericstj about the differences here and he is going to work on factoring these as an extension so the core packaging.props file will remain the same across corefx and wcf. As a secondary change he is also going to clean up the runtime.json difference as well.

**Comment by StephenBonikowsky** [comment]
That's great, thank you!

**Comment by ericstj** [comment]
Refactored this to make the diff for packaging.props identical between the two branches.

File: Packaging.props [hunk 2]
```diff
@@ -18,6 +18,8 @@
     <NativePackagePath>$(MSBuildThisFileDirectory)src/Native/pkg</NativePackagePath>
   </PropertyGroup>
 
+  <Import Project="pkg/baseline/baseline.props" />
```
Comment (weshaggard) [inline Packaging.props]: Can we make this conditional on existence? Again I'm thinking of other repo's like CLI or CoreCLR that might not have a unique baselining requirement.

**Comment by weshaggard** [comment]
Other then the one comment looks good to me.

File: dir.props [hunk 3]
```diff
@@ -90,6 +87,9 @@
     <CoreFxExpectedPrerelease>rc3-24015-00</CoreFxExpectedPrerelease>
   </PropertyGroup>
 
+  <!-- Import packaging props -->
+  <Import Project="$(MSBuildThisFileDirectory)Packaging.props"/>
+
```
Comment (StephenBonikowsky) [inline dir.props]: Does this import need to be dropped lower in the ordering, if so does it also need to be done in corefx?

Comment (ericstj) [inline dir.props]: Yes since it will consume CoreFxExpectedPrerelease.  I can make that change in CoreFx as well.

**Comment by StephenBonikowsky** [comment]
LGTM

File: dir.props (hunk 3 above)
Comment (ericstj) [inline dir.props]: Made this change.

**Comment by ericstj** [comment]
@dotnet-bot test  Innerloop OSX Release Build and Test

**Comment by roncain** [comment]
LGTM.  Doing one CI OuterLoop pass now that we have fixed our infrastructure.

**Comment by roncain** [comment]
@dotnet-bot test outerloop Windows_NT Debug, test outerloop Windows_NT Release

**Comment by roncain** [comment]
LGTM. I'll merge if I don't hear any objections.  Any objections @StephenBonikowsky or @ericstj ?

**Comment by ericstj** [comment]
We need to make sure the corefx change mereges as well, otherwise this will break OTG when it mirrors.

**Comment by roncain** [comment]
Thanks @ericstj .  Do you recommend that whoever merges the CoreFx changes would merge this one instantly after that?  I'm okay with anyone doing the merge when they think the timing is right.

**Comment by ericstj** [comment]
Yes, let's merge at the same time as CoreFx.  I can push the button.

**Pull request merged**